src/tools/RelayInternalTypes.js

```diff
@@ -23,7 +23,7 @@ import type RelayQuery from 'RelayQuery';
 
 export type Call = {
   name: string,
-  type?: string,
+  type?: ?string,
```

Comment:
I haven't thought about this as deeply as you two have, but might we instead want `type` to either be a `string` or not supplied?

```
const orderedCall = {
  name: callOrDirective.name, 
  value,
};
if (metadata && metadata.type) {
  orderedCall.type = metadata.type;
}
orderedCalls.push(orderedCall);
```

That would cut down on the noise in the tests, among other things. If it's important to have `type: null` then, carry on!